nogotofail/mitm/connection/handlers/data/ssl.py

```diff
@@ -203,3 +203,25 @@ def on_ssl(self, client_hello):
                 self.log(logging.ERROR,
                         "Client enabled SSLv3 protocol without TLS_FALLBACK_SCSV")
             self.log_attack_event(data="SSLv3")
+
+
+@handler.passive(handlers)
+class ClientCompressionDetectionHandler(DataHandler):
+    name = "compressiondetection"
+    description = "Detect insecure non-null compression methods in TLS Client Hellos"
+
+    def _handle_compression_detected(self, compression_methods, message):
+            self.log(logging.ERROR, message)
+            self.log_attack_event(data=compression_methods)
+            self.connection.vuln_notify(vuln.VULN_TLS_CLIENT_COMPRESSION)
+
+    def on_ssl(self, client_hello):
+        all_compression_methods = [str(cm) for cm in client_hello.compression_methods]
+        # self.logger.info( "Client compression_methods  %s", all_compression_methods )
+ 
+        for xcm in client_hello.compression_methods:
+            if str(xcm) != 'null':
+                self._handle_compression_detected(all_compression_methods,
+                                               "Client enabled non-null compression methods %s" %
+                                                 (", ".join(all_compression_methods)))
```

Comment:
On line 223 you are ignoring 'null'.  But for the message you appear to be printing all, including null.  Is that your intent?

Reply:
Yes that is intentional; we want to print/log everything client presented.